tests/test_flow.py

```diff
@@ -3517,6 +3576,34 @@ def test_none_key_num_is_not_written():
     assert result == expected
 
 
+def test_field_validator_key():
+    with pytest.raises(ValueError) as value_error:
+        data = pd.DataFrame({"id": [1, 2, 3], "some_data": ["data", "random_data", "my_data"]})
+
+        controller = build_flow(
+            [
+                DataframeSource(dfs=data, key_field="non_existent_column"),
+            ]
+        ).run()
+        controller.await_termination()
+
+    assert str(value_error.value) == "keys ['non_existent_column'] missing from dataframe."
```

Comment:
The common case is to have just one key column, so let's make it user friendly. And please pay attention to the distinction between a key and a key _column_ everywhere.
```suggestion
    assert str(value_error.value) == "key column 'non_existent_column' is missing from dataframe"
```